lib/utils/src/httpclient.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2020 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
This module is the base for common, robust HTTP client logic to be used across all of the CLI, whenever HTTP interaction isn't done through one of the prominent client libs such as aws-sdk-js.

In terms of timeout configuration, retrying, debuggability (logging) and other details, this brings HTTP interaction more or less on par with what we have with our current aws-sdk-js setup.